app/screens/AppNavigator/screens/Balances/BalancesNavigator.tsx

```diff
@@ -27,17 +31,70 @@ export interface BalanceParamList {
   [key: string]: undefined | object
 }
 
+function BalanceActionButton (props: {
+  icon?: React.ComponentProps<typeof MaterialIcons>['name']
+  title?: string
+  onPress: () => void
+  testID: string
+}): JSX.Element {
+  return (
+    <TouchableOpacity
+      testID={props.testID}
+      style={[tailwind('px-2 py-1.5 ml-3 flex-row items-center')]}
+      onPress={props.onPress}
+    >
+      {
+        props.icon !== undefined && (
+          <MaterialIcons name={props.icon} size={20} style={tailwind('text-primary')} />
+        )
+      }
+      {
+        props.title !== undefined && (
+          <Text style={tailwind('mx-1 text-primary font-semibold')}>
+            {translate('screens/BalancesScreen', props.title)}
+          </Text>
+        )
+      }
+    </TouchableOpacity>
+  )
+}
+
+export async function onShare (address: string): Promise<void> {
+  try {
+    await Share.share({
+      message: address
+    })
+  } catch (error) {
+    Logging.error(error.message)
+  }
+}
+
 const BalanceStack = createStackNavigator<BalanceParamList>()
 
 export function BalancesNavigator (): JSX.Element {
+  const { address } = useWalletAddressContext()
+  const navigation = useNavigation()
   return (
     <BalanceStack.Navigator>
       <BalanceStack.Screen
         name='Balances'
         component={BalancesScreen}
         options={{
+          headerTitleAlign: 'center',
           headerTitle: () => <HeaderTitle text={translate('screens/BalancesScreen', 'Balances')} />,
-          headerBackTitleVisible: false
+          headerBackTitleVisible: false,
+          headerLeft: () => (
+            <BalanceActionButton
+              testID='header_share_balance' icon='share'
+              onPress={async () => await onShare(address)}
```

Comment:
Best to have share embedded within `Receive` screen, contextual you are sharing Address. While having share on this screen means you are sharing your balances.